server/events/terraform/terraform_client.go

```diff
@@ -65,6 +65,9 @@ type DefaultClient struct {
 
 	// versionsLock is used to ensure versions isn't being concurrently written to.
 	versionsLock *sync.Mutex
+
+	// usePluginCache determines whether or not to set the TF_PLUGIN_CACHE_DIR env var
+	usePluginCache bool
```

Comment:
Are you going to expose this via a config file flag? Curious as to your use case for why you want to be able to turn this off.

Reply:
I could definitely expose this as a config file flag, this was just to fix the e2e test suite for parallel plans. They would transiently fail with the plugin cache enabled, we never saw this error in production however. The error mentions that the null_resource provider is missing when a plan is run, I wonder if the nature of these tests exacerbates the problem

Reply:
ahh okay